packages/fixie/src/agent.ts

```diff
@@ -480,7 +488,19 @@ export class FixieAgent {
     this.getCodePackage(agentPath);
 
     // Start the agent process locally.
-    FixieAgent.spawnAgentProcess(agentPath, port, environmentVariables);
+    let agentProcess = FixieAgent.spawnAgentProcess(agentPath, port, environmentVariables);
+
+    // Watch files in the agent directory for changes.
+    const watchPath = path.resolve(agentPath);
+    console.log(`🌱 Watching ${watchPath} for changes...`);
+    const watcher = new Watcher(watchPath, { ignoreInitial: true, recursive: true });
+    watcher.on('all', async (event: any, targetPath: string, _targetPathNext: any) => {
+      console.log(`🌱 Restarting local agent process due to ${event}: ${targetPath}`);
+      agentProcess.kill();
+      // Let it shut down gracefully.
+      await new Promise((resolve) => setTimeout(resolve, 2000));
+      agentProcess = FixieAgent.spawnAgentProcess(agentPath, port, environmentVariables);
```

Comment:
Since this isn't triggering a new build, how does the code get updated?

Reply:
And the next question is -- if we did trigger a new build here, do we need to add something to suppress the event if the existing handler is still going so as not to trigger cascading restarts?

Reply:
Somehow I thought the `npx fixie-serve-bin` was triggering a rebuild, but I see that you are correct.
Let me fix this.